File: workspaces/Ruby/oauth.rb
require 'httparty'
require 'base64'

PAYPAL_API_BASE = "https://api.sandbox.paypal.com"
CLIENT_ID = "your_client_id"         # Replace with your client ID
CLIENT_SECRET = "your_client_secret" # Replace with your client secret

def get_access_token
  credentials = Base64.strict_encode64("#{CLIENT_ID}:#{CLIENT_SECRET}")

  headers = {
    "Accept" => "application/json",
    "Authorization" => "Basic #{credentials}",
    "Content-Type" => "application/x-www-form-urlencoded"
  }

  response = HTTParty.post(
    "#{PAYPAL_API_BASE}/v1/oauth2/token",
    headers: headers,
    body: { grant_type: 'client_credentials' }
  )

  data = response.parsed_response
  return data
end

File: workspaces/Dotnet/oauth.cs
using System;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

    public class Program
    {
        public static async Task Main(string[] args)
        {
            var accessToken = await GetAccessToken();
            Console.WriteLine($"Access Token: {accessToken}");
        }

        public static async Task<string> GetAccessToken()
        {
            var PAYPAL_API_BASE = "https://api.paypal.com"; // Replace with the actual base URL
            var CLIENT_ID = "YOUR_CLIENT_ID"; // Replace with your actual client ID
            var CLIENT_SECRET = "YOUR_CLIENT_SECRET"; // Replace with your actual client secret

            var credentials = Convert.ToBase64String(Encoding.UTF8.GetBytes($"{CLIENT_ID}:{CLIENT_SECRET}"));

            using var httpClient = new HttpClient();
            httpClient.BaseAddress = new Uri(PAYPAL_API_BASE);

            var content = new StringContent("grant_type=client_credentials", Encoding.UTF8, "application/x-www-form-urlencoded");
            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", credentials);
            httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

            var response = await httpClient.PostAsync("/v1/oauth2/token", content);
            var responseContent = await response.Content.ReadAsStringAsync();

            if (response.IsSuccessStatusCode)
            {
                // Parse the JSON response to extract the access token
                var accessToken = Newtonsoft.Json.JsonConvert.DeserializeObject<dynamic>(responseContent)?.access_token;
                return accessToken;
            }
            else
            {
                // Handle error scenarios
                // For example, you may throw an exception or return null if the access token retrieval fails
                throw new Exception("Failed to get access token.");
            }
        }
    }

